src/main/java/io/jenkins/plugins/pipelinegraphview/treescanner/PipelineNodeTreeScanner.java

```diff
@@ -126,6 +126,11 @@ public Map<String, List<FlowNodeWrapper>> getAllSteps() {
         for (String stageId : stageNodeMap.keySet()) {
             stageNodeStepMap.put(stageId, getStageSteps(stageId));
         }
+
+        if (stageNodeStepMap.isEmpty()) {
+            stageNodeStepMap.put("(no-stage)", new ArrayList<>(stepNodeMap.values()));
+        }
```

Comment:
this doesn't actually effect the issue like I thought, but it does handle pipelines with no stages nicely